engines/bastion_katello/app/assets/javascripts/bastion_katello/content-hosts/bulk/content-hosts-bulk-system-purpose-modal.controller.js

```diff
@@ -0,0 +1,93 @@
+/**
+ * @ngdoc object
+ * @name  Bastion.content-hosts.controller:ContentHostsBulkSystemPurposeModalController
+ *
+ * @requires $scope
+ * @requires $uibModalInstance
+ * @requires HostBulkAction
+ * @requires Organization
+ * @requires CurrentOrganization
+ * @requires Notification
+ * @requires hostIds
+ *
+ * @description
+ *   A controller for providing bulk action functionality for setting system purpose values
+ */
+angular.module('Bastion.content-hosts').controller('ContentHostsBulkSystemPurposeModalController',
+    ['$scope', '$uibModalInstance', 'HostBulkAction', 'Organization', 'CurrentOrganization', 'Notification', 'hostIds',
+    function ($scope, $uibModalInstance, HostBulkAction, Organization, CurrentOrganization, Notification, hostIds) {
+
+        $scope.organization = Organization.get({id: CurrentOrganization}, function(org) {
+            return org.system_purposes.addons.length;
+        });
+
+        $scope.purposeAddonsList = function () {
+            var addonsList = [];
+
+            if ($scope.organization.system_purposes && $scope.organization.system_purposes.addons.length > 0) {
+                addonsList.concat($scope.organization.system_purposes.addons);
+            }
+            return addonsList;
+        };
```

Comment:
My basic JS knowledge failure here... :smile:  `concat` doesn't modify the array in place; it creates a new array.  So line 28 is evaluated and then the result thrown out, and `addonsList` stays empty.

While we're fixing it may as well simplify even further:
```suggestion
           if ($scope.organization.system_purposes && $scope.organization.system_purposes.addons) {
                return $scope.organization.system_purposes.addons;
            }
            return [];
        };
```